src/jira.ts

```diff
@@ -0,0 +1,2303 @@
+
+type JiraField = {
+	value: string;
+};
+
+interface WebhookBody {
+	timestamp: string;
+	webhookEvent: string;
+	user: Version2Models.DashboardUser;
+	issue: Version2Models.Issue;
+	changelog: Version2Models.Changelog;
+	comment: Version2Models.Comment;
+	transition: Version2Models.Transition & { transitionName: string };
+}
+
+// Routes
+router.post('/webhook', async (req: Request<{}, {}, WebhookBody>, res) => {
+	if (req.query.token !== config.webhookSecret) {
+		res.status(403).end();
+		return;
+	}
+	res.status(200).end();
+
+	if (req.body.webhookEvent && req.body.webhookEvent === 'jira:issue_created') {
+		logger.verbose('New Jira issue webhook triggered');
+
+		let encounteredError = false;
+		const channelLink = await StatusLink.findById(req.body.issue.fields.status.name).lean().exec()
+			.catch((err) => {
+				const eventId = Sentry.captureException(err);
+				logger.error(`Encountered error while finding channel link (${eventId})`);
+				logger.error(err);
+				encounteredError = true;
+			});
+		if (encounteredError) return;
+
+		if (!channelLink) {
+			logger.warn(`No channel link for ${req.body.issue.fields.status.name} found!`);
+			return;
+		}
+
+		const channel = await client.channels.fetch(channelLink.channel)
+			.catch((err) => {
+				const eventId = Sentry.captureException(err);
+				logger.error(`Encountered error while finding Discord channel (${eventId})`);
+				logger.error(err);
+				encounteredError = true;
+			}) as unknown as BaseGuildTextChannel | null;
+		if (encounteredError) return;
+
+		if (channel?.type !== 'GUILD_TEXT') {
+			logger.error(`Channel: ${channelLink.channel} is not a guild text channel`);
+			return;
+		}
+
+		const link = new IdLink({
+			jiraKey: req.body.issue.key,
+			type: 'translation',
+			status: req.body.issue.fields.status.name,
+			// eslint-disable-next-line max-len
+			languages: req.body.issue.fields[config.jira.fields.langs].map((language: JiraField) => language.value),
+			lastUpdate: new Date(),
+			lastStatusChange: new Date(),
+		});
+
+		const embed = new MessageEmbed()
+			.setTitle(`${req.body.issue.key}`)
+			.setColor('#0052cc')
+			.setDescription(req.body.issue.fields.summary ?? 'No description available')
+			.addField('Status', req.body.issue.fields.status.name!)
+			.addField('Source', `[link](${req.body.issue.fields[config.jira.fields.videoLink]})`)
+			.setFooter({ text: `Due date: ${req.body.issue.fields.duedate || 'unknown'}` })
+			.setURL(`${config.jira.url}/projects/${req.body.issue.fields.project.key}/issues/${req.body.issue.key}`);
+
+		const msg = await channel.send({ content: 'A new project is available and can be picked up in Jira', embeds: [embed] })
+			.catch((err) => {
+				const eventId = Sentry.captureException(err);
+				logger.error(`Encountered error while sending message (${eventId})`);
+				logger.error(err);
+				encounteredError = true;
+			});
+		if (encounteredError) return;
+
+		link.discordMessageId = msg!.id;
```

Comment:
```suggestion
		link.discordMessageId = msg?.id;
```
Better be safe than sorry ig. The code can handle missing messages, so not crashing is more important here.